### PR TITLE
* Migrate content of the 'invoice' table

### PR DESCRIPTION
**Pull request by ehuelsmann**

### CODE REVIEW

File: sql/upgrade/sl3.0-1.5.sql [hunk 1]
```diff
@@ -620,23 +620,31 @@ ALTER TABLE ap ENABLE TRIGGER ap_audit_trail;
...
                                     amount, transdate, source,
         CASE WHEN cleared IS NOT NULL THEN TRUE ELSE FALSE END, fx_transaction,
-        memo, approved, cleared, vr_id
-        FROM sl30.acc_trans
-        WHERE chart_id IS NOT NULL AND trans_id IN (
-            SELECT id FROM transactions);
+        memo, approved, cleared, vr_id, invoice.id
+   FROM sl30.acc_trans
+LEFT JOIN sl30.invoice ON acc_trans.id = invoice.id
+  WHERE chart_id IS NOT NULL AND trans_id IN (SELECT id FROM transactions);
```
Comment (ylavoie) [inline sql/upgrade/sl3.0-1.5.sql]: Which trans_id are you referring to? Both invoice and acc_trans have that field to the query is bound to be ambiguous at runtime.

File: sql/upgrade/sl2.8-1.5.sql [hunk 2]
```diff
@@ -458,18 +458,29 @@ ALTER TABLE sl28.acc_trans ADD COLUMN lsmb_entry_id integer;
...
+                               from sl28.chart
+                              where chart.id = acc_trans.chart_id)),
+            amount, transdate, source,
+            CASE WHEN cleared IS NOT NULL THEN TRUE ELSE FALSE END,
+            fx_transaction,
+            memo, approved, cleared, vr_id, invoice.id
+       FROM sl28.acc_trans
+  LEFT JOIN sl28.invoice ON invoice.id = acc_trans.id
   WHERE chart_id IS NOT NULL
         AND trans_id IN (SELECT id FROM transactions);
```
Comment (ylavoie) [inline sql/upgrade/sl2.8-1.5.sql]: Which trans_id are you referring to? Both invoice and acc_trans have that field to the query is bound to be ambiguous at runtime.

File: sql/upgrade/sl3.0-1.5.sql (hunk 1 above)
Comment (ehuelsmann) [inline sql/upgrade/sl3.0-1.5.sql]: They do? Completely missed that. I thought they both have an "id" field, nothing more. Well, I'll have to check in my morning.

File: sql/upgrade/sl3.0-1.5.sql [hunk 3]
```diff
@@ -620,23 +620,32 @@ ALTER TABLE ap ENABLE TRIGGER ap_audit_trail;
...
         CASE WHEN cleared IS NOT NULL THEN TRUE ELSE FALSE END, fx_transaction,
-        memo, approved, cleared, vr_id
-        FROM sl30.acc_trans
-        WHERE chart_id IS NOT NULL AND trans_id IN (
-            SELECT id FROM transactions);
+        memo, approved, cleared, vr_id, invoice.id
+   FROM sl30.acc_trans
+LEFT JOIN sl30.invoice ON acc_trans.id = invoice.id
+                          AND acc_trans.trans_id = invoice.trans_id
+  WHERE chart_id IS NOT NULL AND trans_id IN (SELECT id FROM transactions);
```
Comment (ylavoie) [inline sql/upgrade/sl3.0-1.5.sql]: I think that the WHERE clause needs disambiguation too.

**Pull request merged**